BloodRc PoH Fairy Ring Config

Zeus-Xora opened the pull request:
Need a config for the time being for PoH fairy rings. Can enter the Id manually and PoH teleports should work accross accounts.

Zeus-Xora committed:
Added a config if using PoH to enter PoH Fairy ring ID in Config while I work on a permanent solve. Defaults to ID that has been here, 27097.

Zeus-Xora committed:
Merge remote-tracking branch 'origin/development' into Local

Zeus-Xora committed:
RuneType.java

Zeus-Xora committed:
Poh Fairy ring fix.

Better sleep method for determining when arriving to house.

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/frosty/bloods/BloodsScript.java:
```diff
@@ -444,20 +434,35 @@ private void handleGoingHome() {
+                    return composition.getName().toLowerCase().contains("fairy");
+                })
+                .findFirst().orElse(null);
+
```
We should check if fairyRing is null & use Microbot.log to warn the user that we can't find the fairy ring

Zeus-Xora replied:
Done

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/frosty/bloods/BloodsPlugin.java:
```diff
@@ -47,6 +59,50 @@ BloodsConfig provideConfig(ConfigManager configManager) {
+            }
+            String name = fairyRingComp.getName().toLowerCase();
+            if ((name.contains("fairy ring") || name.contains("spiritual fairy tree")) && Rs2GameObject.isReachable(fairyRing)) {
+                this.fairyRing = fairyRing;
```
Can we remove this fairy ring object logic if we are no longer using it?

Zeus-Xora replied:
I am currently using this in ardougneCloak and in another instance. I was focusing on pohFairyRing for the time being. I will work on removing it if I have the time today

Zeus-Xora committed:
FairyRing null check

Zeus-Xora committed:
FairyRing null check, resets to bank to try again if null.

Should be last commit, apologies

Zeus-Xora committed:
Removed RuneType.java as it is not needed at this time from.
Moved HomeTeleports.java into Teleports.java
Created method for identifying myWorldPoint and added a getter for version in BloodsPlugin.java
Also, in BloodsPlugin.java, removed fairyRing spawn event as not needed.
In BloodsScript.java,defined item IDs with runelites api, changed fairy ring interaction method (for both PoH and Ardougne Cloak), fixed some spelling issues with Microbot.log and added more for users as well as debug, cause wow.
